classes/submission/reviewer/form/ReviewerReviewStep3Form.inc.php

```diff
@@ -34,8 +34,13 @@ function __construct($request, $reviewerSubmission, $reviewAssignment) {
 			return true;
 		}));
 
+		$this->addCheck(new FormValidatorCustom($this, 'recommendation', 'required', 'reviewer.submission.reviewFormResponse.form.recommendationRequired', function($recommendation) {
```

Comment:
I could be wrong, but I think recommendations aren't used in OMP...

Reply:
correct.

So what approach should I use here? Rename to PKPReviewerReviewStep3Form.inc.php and own classes to OMP and OJS, Extend in OJS, leave empty class to OMP?

Reply:
Yeah I think go with that approach. Looks like we already have distinct template files.

Reply:
actually not that easy. The way the code works see here: https://github.com/pkp/pkp-lib/blob/master/pages/reviewer/PKPReviewerHandler.inc.php#L72
So I would probably need to either
1) create app specific handlers for all steps, most of them empty and just calling for pkp-lib version
2) change the PKPReviewerHandler code so that it treats step 3 differently

Reply:
Well another alternative is just to include a hidden `recommendations` field in OMP and set it to `SUBMISSION_REVIEWER_RECOMMENDATION_SEE_COMMENTS`. @asmecher do you have a preference here?

Reply:
I have never liked building class names programmatically -- and neither do IDEs or code analyzers! -- so I would suggest replacing that with something more general, e.g. in the PKP library:
```
public function getReviewForm($step, $request, $reviewerSubmission, $reviewAssignment) {
    switch ($step) {
        case 1: return new ReviewerReviewStep1Form($request, $reviewerSubmission, $reviewAssignment);
        case 2: return new ReviewerReviewStep2Form($request, $reviewerSubmission, $reviewAssignment);
        case 3: return new ReviewerReviewStep3Form($request, $reviewerSubmission, $reviewAssignment);
    }
}
```
Then that could be overridden in the application's subclass.

Reply:
check